src/Event/DataFilterEvent.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+// Copyright (C) 2010-2024, the Friendica project
+// SPDX-FileCopyrightText: 2010-2024 the Friendica project
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+declare(strict_types=1);
+
+namespace Friendica\Event;
+
+/**
+ * Allow Event listener to modify data.
+ */
+final class DataFilterEvent
+{
+	private string $name;
+
+	/** @var string|array */
+	private $data;
+
+	/**
+	 * @param string|array $data
+	 */
+	public function __construct(string $name, $data)
+	{
+		$this->name = $name;
+		$this->data = $data;
+	}
+
+	public function getName(): string
+	{
+		return $this->name;
+	}
+
+	/**
+	 * @return string|array
+	 */
+	public function getData()
+	{
+		return $this->data;
+	}
+
+	/**
+	 * @param string|array $data
+	 *
+	 * @throws \InvalidArgumentException If $data is a string but the original data was an array and vice versa.
+	 */
+	public function setData($data): void
+	{
+		if (is_string($this->data) && is_array($data)) {
```

Comment:
I don't like that we have to check manually for the correct types. Maybe I should create a separate `HtmlFilterEvent`?

Reply:
I assume you just need a light tap on the shoulder to confirm you in this way? Go for it, then. The correct Event type (between HtmlFilterEvent and ArrayFilterEvent) would be determined at the location of the former `Hook::callAll`?

Reply:
I've create a HtmlFilterEvent and have got a great idea: the HtmlFilterEvent documents all possible events as constants that will be than used in the code. This way we have all possible html filters documented and could even deprecate them in the future development. The HookEventBridge will translate the new event names to the old hook names.

I'm working on it, wait for it. :slightly_smiling_face:

Reply:
Reading the Symfony documentation, it seems to be similar to their "KernelEvent" structure with all possible event names as constants in the class. Can't be a bad idea then!